test/PushNotificationTests/PackagedSelfContainedTests.cpp

```diff
@@ -0,0 +1,144 @@
+
+        TEST_CLASS_SETUP(ClassInit)
+        {
+            ::Test::Bootstrap::SetupPackages();
+            return true;
+        }
+
+        TEST_CLASS_CLEANUP(ClassUninit)
+        {
+            ::Test::Bootstrap::CleanupPackages();
+            return true;
+        }
+
+        TEST_METHOD_SETUP(MethodInit)
+        {
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyDataStore());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyLifetimeManager());
+            return true;
+        }
+
+        TEST_METHOD_CLEANUP(MethodUninit)
+        {
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyDataStore());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyLifetimeManager());
+
+            // Need to keep each TEST_METHOD in clean state.
+            try
+            {
+                PushNotificationManager::Default().UnregisterAll();
+            }
+            catch (...)
+            {
+                // We want to unregister regardless of pass/fail to clean the state.
+            }
+            return true;
+        }
+
+        TEST_METHOD(ChannelRequestUsingNullRemoteId)
+        {
+            PushNotificationTestFunctions::ChannelRequestUsingNullRemoteId();
+        }
+
+        TEST_METHOD(ChannelRequestUsingRemoteId)
+        {
+            PushNotificationTestFunctions::ChannelRequestUsingRemoteId();
+        }
+
+        // Currently failing.
+        TEST_METHOD(MultipleChannelClose)
+        {
+            PushNotificationTestFunctions::MultipleChannelClose();
+        }
+
+        TEST_METHOD(VerifyRegisterAndUnregister)
+        {
+            PushNotificationTestFunctions::VerifyRegisterAndUnregister();
+        }
+
+        TEST_METHOD(VerifyRegisterAndUnregisterAll)
+        {
+            PushNotificationTestFunctions::VerifyRegisterAndUnregisterAll();
+        }
+
+        TEST_METHOD(MultipleRegister)
+        {
+            PushNotificationTestFunctions::MultipleRegister();
+        }
+
+        TEST_METHOD(VerifyMultipleRegisterAndUnregister)
+        {
+            PushNotificationTestFunctions::VerifyMultipleRegisterAndUnregister();
+        }
+
+        TEST_METHOD(VerifyMultipleRegisterAndUnregisterAll)
+        {
+            PushNotificationTestFunctions::VerifyMultipleRegisterAndUnregisterAll();
+        }
+
+        TEST_METHOD(VerifyUnregisterTwice)
+        {
+            PushNotificationTestFunctions::VerifyUnregisterTwice();
+        }
+
+        TEST_METHOD(VerifyUnregisterAll)
+        {
+            PushNotificationTestFunctions::VerifyUnregisterAll();
+        }
+
+        TEST_METHOD(VerifyUnregisterAllTwice)
+        {
+            PushNotificationTestFunctions::VerifyUnregisterAllTwice();
+        }
+
+        TEST_METHOD(VerifyUnregisterAndUnregisterAll)
+        {
+            PushNotificationTestFunctions::VerifyUnregisterAndUnregisterAll();
+        }
+
+        TEST_METHOD(VerifyForegroundHandlerSucceeds)
```

Comment:
How would we test for activation outside the handler code path? Are there specific tests for that which need to be moved here?

Reply:
Activation will be handled by a TestApp in another test suite. Looking forward to added improved activation tests soon.